### ISSUE TITLE
Error response doesn't throw actual error

### ISSUE THREAD

**Issue by alimazrae** [yafrica/coinbase-api-v2-java#1]
Whenever Coinbase returns an error, such as 400 - Bad Request or 401 - Not Auth, the error is thrown as a FileNotFoundError with the API call. 

The error code can be seen by setting a breakpoint in CoinbaseHttpResponse getResponse(), however, it only shows the error code, and does not give the actual error returned in the JSON. 

An example of this would be a request with an expired timestamp;
Error thrown; FileNotFoundError
Debugging shows 401 Not Authorised
Online Rest Tester; {"errors":[{"id":"authentication_error","message":"request timestamp expired"}]}

This can make it difficult to debug why the call isn't working. I am trying to send a transaction and am receiving a 400 Bad Request error but can't find out what is wrong with the request.

**Comment by yafrica** [comment]
Hello,
Thanks for your feedback, I don't actually use my own API, but if you want to use it, I can go further.
Could you please validate the fix ?
Regards.
Yoan

**Comment by alimazrae** [comment]
Hi Yoan, 
Thanks for your work on the API, I have integrated it into my final year dissertation project, do I have your permission to use it? It will not be commercialized and I will credit you in my thesis. I have modified it to make it work with Android and changed the OAuth class to calculate the correct signature. 

Regarding this issue, I could not find a definite fix, but by setting up an SSL proxy I can sniff the requests/responses and have identified my problems.

Thanks again, Ali
